qiskit/circuit/quantumcircuit.py

```diff
@@ -1252,6 +1256,7 @@ def append(
         else:
             operation = instruction
         # Convert input to instruction
+        # TODO: define type SupportsInstruction?
```

Comment:
The checks for `hasattr(operation, "to_instruction")` appear in muliple places. We may want to define type which says that class can be converted to `Instruction` instead